pkg/rehearse/jobs_test.go

```diff
@@ -754,3 +756,221 @@ func makeBasePresubmit() *prowconfig.Presubmit {
 		Brancher:     prowconfig.Brancher{Branches: []string{"^master$"}},
 	}
 }
+
+func TestHasChangedTemplateVolume(t *testing.T) {
+	templates := config.CiTemplates{
+		"test-template.yaml": &templateapi.Template{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-template",
+			},
+		},
+		"test-template2.yaml": &templateapi.Template{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-template2",
+			},
+		},
+		"test-template3.yaml": &templateapi.Template{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-template3",
+			},
+		},
+	}
+
+	type expectedToFind struct {
+		exists     bool
+		index      int
+		volumeName string
+	}
+
+	testCases := []struct {
+		description     string
+		jobVolumeMounts []v1.VolumeMount
+		jobVolumes      []v1.Volume
+		expectedToFind  expectedToFind
+	}{
+		{
+			description:     "no volumes",
+			jobVolumeMounts: []v1.VolumeMount{},
+			jobVolumes:      []v1.Volume{},
+			expectedToFind:  expectedToFind{},
+		},
+		{
+			description: "find one in multiple volumes",
+			jobVolumeMounts: []v1.VolumeMount{
+				{
+					Name:      "non-template",
+					MountPath: "/tmp/test",
+				},
+				{
+					Name:      "job-definition",
+					MountPath: "/tmp/test",
+					SubPath:   "test-template.yaml",
+				},
+			},
+			jobVolumes: []v1.Volume{
+				{
+					Name: "job-definition",
+					VolumeSource: v1.VolumeSource{
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{Name: "cluster-e2e-test-template"},
+							Items: []v1.KeyToPath{
+								{Key: "test-template.yaml"},
+							},
+						},
+					},
+				},
+				{
+					Name: "test-volume",
+					VolumeSource: v1.VolumeSource{
+						Projected: &v1.ProjectedVolumeSource{
+							Sources: []v1.VolumeProjection{
+								{
+									Secret: &v1.SecretProjection{
+										LocalObjectReference: v1.LocalObjectReference{Name: "test-secret"},
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Name: "test-volume2",
+					VolumeSource: v1.VolumeSource{
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			expectedToFind: expectedToFind{
+				exists:     true,
+				index:      0,
+				volumeName: "test-template",
+			},
+		},
+		{
+			description: "find one in multiple volumes that for some reason use two templates",
+			jobVolumeMounts: []v1.VolumeMount{
+				{
+					Name:      "non-template",
+					MountPath: "/tmp/test",
+				},
+				{
+					Name:      "job-definition",
+					MountPath: "/tmp/test",
+					SubPath:   "test-template.yaml",
+				},
+			},
+			jobVolumes: []v1.Volume{
+				{
+					Name: "job-definition",
+					VolumeSource: v1.VolumeSource{
```

Comment:
you could make a helper  `func volumeSource(name, key string) v1.VolumeSource` to make these tests more readable